File: lib/models/city.dart
class City {
  final int id;
  final String name;
  final int countryId;
  final String countryCode;
  final int population;
  final String? countryName;
  final double? latitude;
  final double? longitude;
  final String? timezone;
  final String? currency;
  final String? currencySymbol;
  final String? phoneCode;
  final String? languages;
  final String? countryEmoji;

  City({
    required this.id,
    required this.name,
    required this.countryId,
    required this.countryCode,
    required this.population,
    this.countryName,
    this.latitude,
    this.longitude,
    this.timezone,
    this.currency,
    this.currencySymbol,
    this.phoneCode,
    this.languages,
    this.countryEmoji,
  });

  factory City.fromJson(Map<String, dynamic> json) {
    return City(
      id: int.tryParse(json['id']?.toString() ?? '0') ?? 0,
      name: json['name'] ?? '',
      countryId: int.tryParse(json['country_id']?.toString() ?? '0') ?? 0,
      countryCode: json['country_code'] ?? '',
      population: int.tryParse(json['population']?.toString() ?? '0') ?? 0,
      countryName: json['country_name'],
      latitude: json['latitude'] != null
          ? double.tryParse(json['latitude'].toString())
          : null,
      longitude: json['longitude'] != null
          ? double.tryParse(json['longitude'].toString())
          : null,
      timezone: json['timezone'],
      currency: json['currency'],
      currencySymbol: json['currency_symbol'],
      phoneCode: json['country_phonecode'],
      languages: json['country_languages'],
      countryEmoji: json['country_emoji'],
    );
  }
}

File: city.php
<?php
header('Content-Type: application/json');
header('Access-Control-Allow-Origin: *');
header('Access-Control-Allow-Methods: GET, POST, OPTIONS');
header('Access-Control-Allow-Headers: Content-Type');

$host = 'localhost';
$user = 'root';
$pass = '';
$db = 'world_db';

$conn = new mysqli($host, $user, $pass, $db);

if ($conn->connect_error) {
    echo json_encode(['error' => 'Connection failed']);
    exit;
}

$id = isset($_GET['id']) ? intval($_GET['id']) : 0;

if ($id == 0) {
    echo json_encode(['error' => 'City ID required']);
    exit;
}

$sql = "SELECT 
            c.id,
            c.name as city_name,
            c.country_id,
            c.country_code,
            c.latitude,
            c.longitude,
            c.population,
            c.timezone,
            co.name as country_name,
            co.emoji as country_emoji,
            co.currency,
            co.currency_symbol,
            co.phonecode as country_phonecode
        FROM cities c
        JOIN countries co ON c.country_id = co.id
        WHERE c.id = $id";

$result = $conn->query($sql);

if ($result && $result->num_rows > 0) {
    $city = $result->fetch_assoc();
    echo json_encode($city, JSON_UNESCAPED_UNICODE);
} else {
    echo json_encode(['error' => 'City not found']);
}

$conn->close();
?>